sklearn/decomposition/_dict_learning.py

```diff
@@ -909,6 +909,9 @@ def _transform(self, X, dictionary):
         SparseCoder."""
         X = self._validate_data(X, reset=False)
 
+        if hasattr(self, "alpha") and self.transform_alpha is None:
+            self.transform_alpha = self.alpha
```

Comment:
We don't modify the init parameters. You should set `transform_alpha = self.alpha` and then use transform_alpha. 

Since for now we don't change the behavior, you can write `transform_alpha = 1` to prepare for the future change.